core/src/datayoga_core/blocks/relational/write/block.py

```diff
@@ -69,8 +69,9 @@ async def run(self, data: List[Dict[str, Any]]) -> BlockResult:
                     Result(status=Status.REJECTED, payload=record, message=f"unknown opcode '{opcode}'")
                     for record in opcode_groups[opcode]
                 ])
-            self.execute_upsert(opcode_groups[OpCode.CREATE] + opcode_groups[OpCode.UPDATE])
-            self.execute_delete(opcode_groups[OpCode.DELETE])
+
+            self.execute_upsert(self.explode_if_specified(opcode_groups[OpCode.CREATE] + opcode_groups[OpCode.UPDATE]))
```

Comment:
why not do the explode in the beginning before the grouping?

Reply:
As explained, we first check that the opcode field exists and valid and if not reject such records. So it's inefficient to explode the records and then to reject those while we can do it in advance.